src/jaspObject.cpp

```diff
@@ -126,6 +126,96 @@ void jaspObject::removeChild(jaspObject * child)
 	child->parent = NULL;
 }
 
+Json::Value jaspObject::getObjectFromNestedOption(std::vector<std::string> nestedKey, Json::Value ifNotFound) const
+{
+	Json::Value obj = currentOptions;
+	for (const auto& key: nestedKey)
+	{
+		// NOTE: this fails if we have options which are an array where some elements are named, but not all.
+		// I think that would violate the Json spec for arrays, but I'm not 100% sure.
```

Comment:
I dont think we can have an array with names in C++, `Json::array != Json::object` luckily

Reply:
good to know! I wasn't sure because this is possible in R.